test/integration/admin/api/member_permissions_controller_test.rb

```diff
@@ -6,200 +6,205 @@ class Admin::Api::MemberPermissionsControllerTest < ActionDispatch::IntegrationT
 
     put admin_api_permissions_path(id: another_user.id, format: :json), params: params
 
     assert_response :forbidden
   end
 
   test "PUT: setting an invalid allowed section" do
-    @user.update({ allowed_sections: ['partners'], allowed_service_ids: [@services.first.id] })
-    params = { allowed_sections: ['invalid'], access_token: @token }
+    user.update({ allowed_sections: ['partners'], allowed_service_ids: [service1_id] })
+    params = { allowed_sections: ['invalid'], access_token: token }
 
-    put admin_api_permissions_path(id: @user.id, format: :json), params: params
+    put admin_api_permissions_path(id: user.id, format: :json), params: params
 
     assert_response :success
     assert_not_nil(permissions = JSON.parse(response.body)['permissions'])
     assert_empty permissions['allowed_sections']
-    assert_equal [@services.first.id], permissions['allowed_service_ids']
+    assert_equal [service1_id], permissions['allowed_service_ids']
 
-    @user.member_permissions.reload
-    assert_empty @user.allowed_sections.to_a
-    assert_equal [@services.first.id], @user.allowed_service_ids
+    user.member_permissions.reload
+    assert_empty user.allowed_sections.to_a
+    assert_equal [service1_id], user.allowed_service_ids
   end
 
   test "PUT: one of the allowed section is invalid" do
-    @user.update({ allowed_sections: ['partners'], allowed_service_ids: [@services.first.id] })
-    params = { allowed_sections: %w[invalid settings], access_token: @token }
+    user.update({ allowed_sections: ['partners'], allowed_service_ids: [service1_id] })
+    params = { allowed_sections: %w[invalid settings], access_token: token }
 
-    put admin_api_permissions_path(id: @user.id, format: :json), params: params
+    put admin_api_permissions_path(id: user.id, format: :json), params: params
 
     assert_response :success
     assert_not_nil(permissions = JSON.parse(response.body)['permissions'])
     assert_equal ['settings'], permissions['allowed_sections']
-    assert_equal [@services.first.id], permissions['allowed_service_ids']
+    assert_equal [service1_id], permissions['allowed_service_ids']
 
-    @user.member_permissions.reload
-    assert_equal [:settings], @user.allowed_sections.to_a
-    assert_equal [@services.first.id], @user.allowed_service_ids
+    user.member_permissions.reload
+    assert_equal [:settings], user.allowed_sections.to_a
+    assert_equal [service1_id], user.allowed_service_ids
   end
 
   test "PUT: setting services, when some are non-existent only enables existent ones" do
-    @user.update({ allowed_sections: ['partners'], allowed_service_ids: [@services.first.id] })
-    params = { allowed_service_ids: [[@services.last.id.to_s],'22'], access_token: @token }
+    user.update({ allowed_sections: ['partners'], allowed_service_ids: [service1_id] })
+    params = { allowed_service_ids: [[service2_id.to_s], nonexistent_id.to_s], access_token: token }
```

Comment:
I know the original was like this, but why do we pass an array of an array and a single element instead of just two normal string elements?

Reply:
Oops, yeah, that doesn't make sense :upside_down_face: I think it's a bug, and I'll fix it.

Reply:
I was trying to figure out why that worked in the first place, and it seems that it was actually a bug: https://github.com/rails/rails/issues/38698
It was fixed here: https://github.com/rails/rails/pull/48148 But I can't seem to find an easy way to know which release it is in, but I assume it's in Rails 7 in any case.

Good catch @akostadinov !